sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -138,6 +143,9 @@ class PrioritySemaphore[T](val maxPermits: Long)
   }
 
   private def canAcquire(numPermits: Long): Boolean = {
-    occupiedSlots + numPermits <= maxPermits
+    val hasPermits = occupiedSlots + numPermits <= maxPermits
+    val withinTaskLimit = maxConcurrentGpuTasksLimit < 0 || 
```

Comment:
I think this should be maxConcurrentGpuTask <= 0. Otherwise, if I set the max limit to 0, then nothing gets scheduled _ever_. 

That or make the default 0, and 0 means it's no limit. It feels that 0 may be more natural.